### ISSUE TITLE
Hash on whole record

### ISSUE THREAD

**Issue by labou77** [mtakemi/fluent-plugin-genhashvalue#3]
Hello
I would like to propose a modification to be able compute the hash on the complete record object instead of having to choose a combination of keys. That would make the usage of this filter independent of any index model. 
To compute on the whole record, we could either pass an empty keys array or we could add a new input parameter.
I locally tried the following to replace this: `s += keys.map {|k| record[k]}.join(separator)` by  `s += record.to_s` for a quick test and it is working.
Is that feasible?
Thanks

**Comment by ghost** [comment]
does this modification work for both cases i.e., hash on select keys and on entire record?

**Comment by labou77** [comment]
Yes. I have implemented it on a fork project : https://github.com/Labou77Code/fluent-plugin-genhashvalue-alt
